plugins/meta/portmap/portmap_nftables.go

```diff
@@ -110,23 +111,23 @@ func (pmNFT *portMapperNFTables) forwardPorts(config *PortMapConf, containerNet
 	})
 
 	tx.Add(&knftables.Chain{
-		Name:     "input",
+		Name:     "prerouting",
 		Type:     knftables.PtrTo(knftables.NATType),
-		Hook:     knftables.PtrTo(knftables.InputHook),
+		Hook:     knftables.PtrTo(knftables.PreroutingHook),
```

Comment:
OK, I think I know what I was confused about: in `prerouting` it hasn't yet decided where it's going to route the packet to, so you can't use `oifname`. I was thinking that meant you couldn't use `fib daddr type` either, but that's wrong; `fib daddr type` answers "what does the routing table say we _should_ do with this packet?", not "what are we _actually_ going to do with this packet?", so it doesn't depend on the routing decision.